app/routes/public/cfs/index.js

```diff
@@ -5,6 +5,15 @@ export default Route.extend({
     return this.get('l10n').t('Call for Speakers');
   },
 
+  async beforeModel(transition) {
+    let hash = transition.params['public.cfs'].speaker_call_hash;
+    const eventDetails = this.modelFor('public');
+    const speakersCall = await eventDetails.get('speakersCall');
+    if (speakersCall.hash !== hash) {
+      this.transitionTo('not-found');
```

Comment:
This should only be done in the following cases.

- CFS Privacy setting is private and the hash is either empty or wrong
- Hash is not empty and incorrect

---

The following should show the CFS page to the user

- CFS is public and no hash is entered
- CFS is public and a valid hash is entered
- CFS is private and a valid hash is entered

Reply:
`/cfs` doesn't get mapped to 
```javascript
 this.route('cfs', { path: '/cfs/:speaker_call_hash' }, function() {
      this.route('new-speaker');
      this.route('new-session');
    });
```
I get a 404 instead, so I was passing the hash in the public url but that is major security breach as pointed out. Couldn't find any resource where one route was mapped to two different urls

Reply:
Is it okay if I do this - 
```javascript
this.route('cfs', function() {
      this.route('new-speaker');
      this.route('new-session');
    });
this.route('cfs', { path: '/cfs/:speaker_call_hash' }, function() {
      this.route('new-speaker');
      this.route('new-session');
    });
```
https://stackoverflow.com/questions/24923120/emberjs-using-multiple-paths-urls-for-one-route?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Reply:
Hey @mayank8318 ... Sorry for the late reply. Yep. You can implement it like this. This is fine